### PR TITLE
Enable dark mode based on user OS settings

### PR DESCRIPTION
**Pull request by FelixMalfait**
- Take the theme from browser's preference
- Adapt Storybook to have a dark mode toggle

Todo later: store theme config on backend to allow user to overwrite system settings with light/dark theme

### CODE REVIEW

**Comment by FelixMalfait** [comment]
Not sure I understand the UI diff in storybook? (checkbox icons!?)

**Comment by charlesBochet** [comment]
> Not sure I understand the UI diff in storybook? (checkbox icons!?)

Yes, I have seen this diff on all other PRs too, not sure where it's coming from, let's ignore it for now as we learn how to use chromatic

**Review by charlesBochet — CHANGES_REQUESTED** [review]
PR is nice but we can't merge it as this will expose dark theme and dark theme is not great right now. We can do a small pass on dark theme to fix issues and then merge it

File: front/src/index.tsx [hunk 1]
```diff
@@ -5,9 +5,13 @@ import App from './App';
 import { BrowserRouter } from 'react-router-dom';
 import { ApolloProvider } from '@apollo/client';
 import '@emotion/react';
-import { ThemeType } from './layout/styles/themes';
 import { apiClient } from './apollo';
 import { RecoilRoot } from 'recoil';
+import { darkTheme, lightTheme, ThemeType } from './layout/styles/themes';
+import { ThemeProvider } from '@emotion/react';
+
+const isDarkMode = window.matchMedia('(prefers-color-scheme: dark)');
```
Comment (charlesBochet) [inline front/src/index.tsx:13]: would be nice to extract this in utils

File: front/src/layout/AppLayout.tsx [hunk 2]
```diff
@@ -28,12 +26,10 @@ type OwnProps = {
 
 function AppLayout({ children, user }: OwnProps) {
   return (
-    <ThemeProvider theme={lightTheme}>
```
Comment (charlesBochet) [inline front/src/layout/AppLayout.tsx:31]: I think we need to keep ThemeProvider at AppLayout level as we will need to get it from the user at long term, we cannot do that from index.ts

**Review by charlesBochet — CHANGES_REQUESTED** [review]

File: front/src/layout/AppLayout.tsx (hunk 2 above)
Comment (FelixMalfait) [reply front/src/layout/AppLayout.tsx:31]: Yes I tried doing so initially but then it was harder to have the entire App work well in storybook. People/Companies pages worked well but if you put the theme in Layout then the storybook toggle will be ignored. I can give it another shot

File: front/src/index.tsx (hunk 1 above)
Comment (FelixMalfait) [reply front/src/index.tsx]: Done

**Comment by charlesBochet** [comment]
I think we should close it as conflicts will be almost impossible to resolve here

**Pull request closed without merging**